changelog.d/15773.feature

```diff
@@ -0,0 +1 @@
+Allow configuring the set of workers to proxy outbound federation traffic through via `outbound_federation_restricted_to`.
```

Comment:
@erikjohnston Mind giving this a review pass before we pass it to an unsuspecting reviewer without context?

I think it's good to go but want to wait for https://github.com/matrix-org/synapse/pull/15806 to merge at the very least to reduce the diff clutter here.

The commits could be rebased to be nice but then we lose some historical context that is nice to reference on how you had things before I came in. Opting to just leave it